public/components/model_list/index.tsx

```diff
@@ -137,10 +137,18 @@ export const ModelList = ({ notifications }: { notifications: CoreStart['notific
             pagination={pagination}
             onChange={handleTableChange}
             onModelNameClick={handleViewModelDrawer}
+            onModelDeleteClick={handleModelDelete}
             onResetClick={handleReset}
             error={!!error}
           />
-          <ModelConfirmDeleteModal ref={confirmModelDeleteRef} onDeleted={handleModelDeleted} />
+          {deployedVersion.length === 0 ? (
```

Comment:
I think this judgement can be removed, we can add one more ref to store the deployment modal.